rapids-cmake/cython-core/cython.cmake

```diff
@@ -0,0 +1,213 @@
+      set(${option_var_name} ${option_flag})
+    else()
+      set(${option_var_name} "")
+    endif()
+  endforeach()
+
+  # Validate that only a single PY argument was provided
+  # TODO: Do we need to check this, or can we rely on the underlying `cython`
+  # invocation to provide a sufficiently meaningful error here?
+  set(num_py_args 0)
+  if(${_RAPIDS_COMPILE_PY2})
+    math(EXPR num_py_args "${num_py_args} + 1")
+  endif()
+  if(${_RAPIDS_COMPILE_PY3})
+    math(EXPR num_py_args "${num_py_args} + 1")
+  endif()
+  if(${_RAPIDS_COMPILE_PY3STR})
+    math(EXPR num_py_args "${num_py_args} + 1")
+  endif()
+  
+  if(num_py_args GREATER 1)
+    message(FATAL_ERROR "Only one of PY2, PY3, and PY3STR may be provided")
+  endif()
+  
+  # Handle one-value args
+  list(LENGTH _rapids_cython_one_value num_one_value)
+  math(EXPR num_one_value "${num_one_value} - 1")
+  foreach(i RANGE ${num_one_value})
+    list(GET _rapids_cython_one_value ${i} option_name)
+    list(GET _rapids_cython_one_value_vars ${i} option_var_name)
+
+    # If the parameter was provided without a value, check if Cython provides
+    # a default. If so, pass the flag without arguments, otherwise raise an
+    # error.
+    if(option_name IN_LIST _RAPIDS_COMPILE_KEYWORDS_MISSING_VALUES)
+      if (option_name IN_LIST _rapids_cython_one_value_has_default)
+        list(GET _rapids_cython_one_value_flags ${i} option_flag)
+        set(${option_var_name} ${option_flag})
+      else()
+        message(FATAL_ERROR "Option ${option_name} must be provided with a single argument")
+      endif()
+    # If the parameter was provided with the value, set it correctly.
+    elseif(DEFINED "_RAPIDS_COMPILE_${option_name}")
+      list(GET _rapids_cython_one_value_flags ${i} option_flag)
+      # TODO: Based on the CLI help the --embed option seems to use a syntax
+      # with an equals sign (--embed=X rather than --embed X). If that is
+      # correct, may need a small special-casing here.
+      set(${option_var_name} "${option_flag} ${_RAPIDS_COMPILE_${option_name}}")
+    endif()
+  endforeach()
+
+  # Now handle the multi-value args. These generally need different treatments
+  if(NOT DEFINED _RAPIDS_COMPILE_SOURCE_FILES)
+    message(FATAL_ERROR "You must provide a list of source files to compile")
+  endif()
+
+  set(include_dirs)
+  if(DEFINED _RAPIDS_COMPILE_INCLUDE_DIRS)
+    foreach(dir IN LISTS _RAPIDS_COMPILE_INCLUDE_DIRS)
+      set(include_dirs "${include_dirs} -I ${dir}")
+    endforeach()
+  endif()
+
+  set(directives)
+  if(DEFINED _RAPIDS_COMPILE_DIRECTIVES)
+    list(JOIN _RAPIDS_COMPILE_DIRECTIVES "," directives)
+    set(directives "-X ${directives}")
+  endif()
+
+  set(compile_time_env)
+  if(DEFINED _RAPIDS_COMPILE_COMPILE_TIME_ENV)
+    list(JOIN _RAPIDS_COMPILE_COMPILE_TIME_ENV "," compile_time_env)
+    set(compile_time_env "-X ${compile_time_env}")
+  endif()
+
+  # Combine arguments into a usable string
+  set(args)
```

Comment:
The two foreach loops can be simplified to:
```cmake
set(args)
foreach(arg IN LISTS _rapids_cython_option_vars _rapids_cython_one_value_vars)
  if(${arg})
    list(APPEND args "${${arg}}")
  endif()
endforeach()
```

Reply:
Good call, thanks!